sdks/java/core/src/main/java/org/apache/beam/sdk/io/WriteFiles.java

```diff
@@ -394,6 +418,7 @@ public void startBundle(StartBundleContext c) {
 
     @ProcessElement
     public void processElement(ProcessContext c, BoundedWindow window) throws Exception {
+      sink.getDynamicDestinations().setSideInputAccessorFromProcessContext(c);
```

Comment:
Alternatively we could make SideInputAccessor a parameter to every method on DynamicDestinations. I believe we considered this when we did BigQueryIO and rejected it. @jkff might remember why we rejected it. (possibly because it seemed like it made the API awkward to always pass a parameter the user might not use).